x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseService.java

```diff
@@ -439,7 +453,12 @@ void startBasicLicense(PostStartBasicRequest request, final ActionListener<PostS
             "start basic license",
             listener
         );
-        clusterService.submitStateUpdateTask(task.getDescription(), task, newExecutor());
+        clusterService.submitStateUpdateTask(
+            task.getDescription(),
+            task,
+            ClusterStateTaskConfig.build(Priority.NORMAL, request.masterNodeTimeout()),
```

Comment:
Similarly here: the `?master_timeout` was read in `RestPostStartBasicLicense` but we used to ignore it.